File: leehyowonzero/2week/12915.py
def solution(strings, n):
    strings.sort() # n번 인덱스가 동일 시 사전순으로 정렬하기 위하여 미리 사전순 정렬 실시
    strings.sort(key = lambda x : x[n]) # 문제에서 요구하는 n번 인덱스 기준 정렬
    return strings

File: jekyun-park/week2/programmers/12915.swift
//
//  main.swift
//  SwiftAlgorithms
//
//  Created by 박제균 on 2022/01/06.
//

import Foundation

func solution(_ strings:[String], _ n:Int) -> [String] {
    
    var dict : [Character:[String]] = [:]
    
    for string in strings {
        let char = string[String.Index(encodedOffset: n)]
        
        if (dict[char] == nil) {
            dict[char] = []
            dict[char]?.append(string)
        } else {
            dict[char]?.append(string)
        }
    }
    
    let sortedDict = dict.sorted { a, b in
        return (a.key < b.key)
    }
    var answer :[String] = []
    
    for (key,value) in sortedDict {
        
        answer += value.sorted()
    }
    
    return answer
}

// refactor

func solution2(_ strings:[String], _ n:Int) -> [String] {
    return strings.sorted { Array($0)[n] == Array($1)[n] ? $0 < $1 : Array($0)[n] < Array($1)[n] }
}

